socceraction/spadl/opta.py

```diff
@@ -874,20 +877,159 @@ def extract_competitions(self) -> Dict[int, Dict[str, Any]]:
         )
         return {season_id: season}
 
+    def extract_teams(self) -> Dict[int, Dict[str, Any]]:
+        match_info = self.get_match_info()
+        contestants = assertget(match_info, 'contestant')
+        teams = {}
+        for contestant in contestants:
+            team_id = assertget(contestant, 'id')
+            team = dict(
+                team_id=team_id,
+                team_name=assertget(contestant, 'name'),
+            )
+            teams[team_id] = team
+        return teams
+
+    def extract_players(self) -> Dict[int, Dict[str, Any]]:
+        live_data = self.get_live_data()
+        events = assertget(live_data, 'event')
+
+        game_duration = 90
+        players = {}
+
+        players_data = {
+            'position_code': [],
```

Comment:
What is "position_code"? Does this correspond to the "starting_position_id" field in the `OptaPlayerSchema`? If so, please rename it. Otherwise, add a new field to the `StatsPerformPlayerSchema`.

Reply:
I don't know what `starting_position_id` means by default because the schema is without description.

`position_code` is number from 1 (goalkeeper) to 5 (forward). If it equals to `starting_position_id` I will change.

Reply:
`starting_position_id` is the internal ID used by the data providers to refer to a specific position. Based on your description, I think the "position_in_formation" field would fit this.